ssl/quic/quic_rx_frame.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <internal/packet.h>
+#include <internal/nelem.h>
+#include <internal/quic_wire.h>
+/* ASSUMPTION: this is the OSSL_QRL header file */
+#include <internal/quic_record_layer.h>
+
+static int depack_collect_ack_data(PACKET *pkt, OSSL_ACKM_RX_PKT *ackm_data)
+{
+    /* Initialize by pretending the packet isn't ACK eliciting */
+    ackm_data->is_ack_eliciting = 0;
+
+    while (PACKET_remaining(&pkt) > 0) {
+        if (!ossl_quic_wire_peek_frame_header(&pkt, &frame_type))
```

Comment:
This looks like it will infinite loop. `ossl_quic_wire_peek_frame_header` only does a peek so does not advance the `PACKET`....so `PACKET_remaining()` will always be ` > 0`

Reply:
Right...  unfortunately there's no comfortable function to skip a whole frame regardless of its type.
I can add one.

Reply:
> I can add one.

Actually no, that wasn't quite as easy as I thought...  looking more closely, it seems more sane to have handler modes.  Ugly, but better than duplicating the switch into two functions...

Reply:
Rewrite done